packages/runner/src/runner.ts

```diff
@@ -174,24 +187,21 @@ export class Runner<X extends AppConfig> implements IComponent {
         this.inputDataStream = new DataStream();
 
         // do not await here, allow the rest of initialization in the caller to run
-        readInputStreamHeaders(this.inputStream!)
-            .then(headers => {
-                const contentType = headers["content-type"];
+    }
 
-                this.logger.log(`Content-Type: ${contentType}`);
+    //readInputStreamHeaders(this.inputStream!)
+    async setInputContentType(headers: any) {
+        const contentType = headers["content-type"];
 
-                mapToInputDataStream(this.inputStream!, contentType).pipe(this.inputDataStream!);
-            }).catch(e => {
-                this.logger.error("Error in input stream");
-                this.logger.error(e);
-            // @TODO think about how to handle errors in input stream
-            });
+        this.logger.log(`Content-Type: ${contentType}`);
+
+        mapToInputDataStream(this.inputStream!, contentType).pipe(this.inputDataStream!);
     }
 
     async hookupOutputStream() {
         this.outputStream = createWriteStream(this.outputFifoPath);
         this.outputDataStream
-            .JSONStringify()
+            //JSONStringify()
```

Comment:
TBD

Reply:
This should be defined by:

1) Package.json
2) Return value of the last item in sequence

Where 2 overrides 1.

Reply:
thanks, we certainly need to handle it but we need a consultation with you about it. It will also cause changes that need to be well tested. Therefore, we propose to solve it in a separate PR:
https://app.asana.com/0/1199599848882722/1200949562077777